core/drand_public.go

```diff
@@ -88,37 +88,35 @@ func (d *Drand) PublicRandStream(req *drand.PublicRandRequest, stream drand.Publ
 	}
 	addr := net.RemoteAddress(stream.Context())
 	done := make(chan error, 1)
+	send := func(b *chain.Beacon) bool {
+		err := stream.Send(beaconToProto(b))
+		err2 := stream.Context().Err()
+		if err != nil || err2 != nil {
+			d.log.Error("stream_to", addr, "stop", err)
+			done <- err
```

Comment:
Can't this hang forever because the best `send(bb)` errors and then the the `beacon.AddCallback()` may potentially trigger another `send(b)` which can also error?

Also, if the context expired, error put on the done channel will be nil. Is that what we want?

Reply:
Outdated code now.